accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -1112,7 +1115,7 @@ The section 'Scripts' in the language specification is replaced by the
 following:
 
 Let _L_ be a library that exports a declaration _D_ named `main`.  It is a
-compile-time error unless _D_ is a function declaration.  It is a
+compile-time error unless _D_ is a non-getter function declaration.  It is a
```

Comment:
Though not strictly necessary, we could also say 'non-getter and non-setter function declaration', in order to outlaw the confusing permission to have `set main(_) {}`.

Reply:
This change isn't *technically* needed. A "function declaration" is a concept separate from a "getter declaration". I do believe the spec occasionally refers to getters as functions, though, so it's probably better to double down on the redundancy overhead.

Reply:
Thanks!

We actually have the following in section `Function Declarations`: `A function declaration consists of an identifier indicating the function’s name, possibly prefaced by a return type. The function name is followed by a signature and body. For getters, the signature is empty.` So a function declaration can be a getter declaration, just like a function can be a getter.

But what do you think about outlawing the _setter_? It is confusing that it is allowed, but its name is of course `main=`, so there is no harm in allowing it.

Reply:
Let's take the setter separately, to get this one settled.